pkg/image/registry/image/strategy.go

```diff
@@ -112,6 +112,37 @@ func (imageStrategy) clearSignatureDetails(image *api.Image) {
 	}
 }
 
+// signatureStrategy implements behaviour for signatures subset of image.
+type signatureStrategy struct {
+	imageStrategy
+}
+
+// SignatureStrategy is the default logic that applies when updating
+// image' signatures via the REST API.
+var SignatureStrategy = signatureStrategy{Strategy}
+
+// Canonicalize normalizes the object after validation.
+func (signatureStrategy) Canonicalize(obj runtime.Object) {
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+// It allows to update only signatures field of the image.
+func (s signatureStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	oldImage := old.(*api.Image)
+	image := obj.(*api.Image)
+
+	image.Labels = oldImage.Labels
+	// Allow to modify Annotations
```

Comment:
No, signatures should only affect signatures.  That means building this the other way around.

I think this code gets less error prone if we change the `images/signatures` to take *only* a signature.